### PR TITLE
Prepare release 1.4.0

### PR DESCRIPTION
**Pull request by dgarros**
Mainly add changlog and update version in pyproject.toml
I hope i didn't miss a PR from 1.3.0, would be good to update the changelog along with the PRs in the future

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,29 @@
...
+### Added
+
+- #83 - Add support for Python 3.10
+- #59 - Add proper documentation published in Read the doc
+- #70 - Add `add_or_update()` method to DiffSync class that requires a DiffSyncModel to be passed in and will attempt to add or update an existing object
+- #80 - Add support for an existing Diff object to be passed to `sync_to()` & `sync_from()` to prevent another diff from being calculated.
+- #87 - Add new model flags : `SKIP_UNMATCHED_BOTH`, `SKIP_UNMATCHED_SRC` & `SKIP_UNMATCHED_DST` to match the behavior of the global flags
+- #81 - Add a new example based on PeeringDB
+- #53 - Add a new example based on pynautobot and Nautobot REST API
+- #75 - Add support for Structlog v21 in addition to v20.
```
Comment (glennmatthews) [inline CHANGELOG.md:14]: Can we put these in numerical order? The current order seems random to me.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,29 @@
...
+
+- #69 - Replace Travis CI with Github Actions to run unit tests
+- #43 - `get()` and `get_by_uids()` now raises an `ObjectNotFound` exception, instead of silently skipping over unmatched uids
+- #90 - Convert list of actions (`create`, `update`, `delete`) to proper Enum
+- #82 - Update lock file with latest versions.
+
+### Fixed
+
+- #51 - Update minimum Pydantic version due to security advisory GHSA-5jqp-qgf6-3pvh
+
```
Comment (glennmatthews) [inline CHANGELOG.md:26]: Looks like may be missing #62, #63, #68, and #72 here. Agreed that we need to stay on top of this in the future.

File: CHANGELOG.md (hunk 1 above)
Comment (dgarros) [reply CHANGELOG.md:14]: I tried to bundle the update by topics ... as much as possible
I did it sequentially initially but I found it harder to read ... happy to change

File: CHANGELOG.md (hunk 2 above)
Comment (dgarros) [reply CHANGELOG.md:26]: I purposely skipped these as they seam to be minor changes related to other PRs we already have in the changelog
I could add them on the same line or would you rather have a dedicated line per PR ?

File: CHANGELOG.md (hunk 1 above)
Comment (glennmatthews) [reply CHANGELOG.md:14]: Personally I'd prefer sequential order just for sake of consistency. Thanks.

File: CHANGELOG.md (hunk 2 above)
Comment (glennmatthews) [reply CHANGELOG.md:26]: OK, I was just referencing your comment about hoping not to have missed any. :-) Personally I'd rather be exhaustive in the list of included PRs, and if necessary have a separate section specifically calling out user-facing changes if the list of PRs is too much?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,33 @@
...
+- #72 - Add core engine section in docs and rename example directories
+- #75 - Add support for Structlog v21 in addition to v20.
+- #80 - Add support for an existing Diff object to be passed to `sync_to()` & `sync_from()` to prevent another diff from being calculated.
+- #81 - Add a new example based on PeeringDB
+- #83 - Add support for Python 3.10
+- #87 - Add new model flags : `SKIP_UNMATCHED_BOTH`, `SKIP_UNMATCHED_SRC` & `SKIP_UNMATCHED_DST` to match the behavior of the global flags
+
+### Changed
+
+- #43 - `get()` and `get_by_uids()` now raises an `ObjectNotFound` exception, instead of silently skipping over unmatched uids
```
Comment (glennmatthews) [inline CHANGELOG.md:20]: #43 was in 1.1.0, it's not a new change for 1.4.0.
```suggestion
```

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**